### DISCUSSION TITLE
What is the expected loss range during fine-tuning?

### DISCUSSION THREAD

**Discussion by bruceLogic** [QwenLM/Qwen3-TTS#252] [Q&A]
Hi, thanks for this great project!

I'm fine-tuning Qwen3-TTS using sft_12hz.py and would like to ask about the expected loss values during training.

From the code, the total loss is computed as:

loss = outputs.loss + 0.3 * sub_talker_loss

Where outputs.loss is the CrossEntropyLoss for codec layer 0 (main talker), and sub_talker_loss is the CrossEntropyLoss for codec layers 1–15 (sub-talker / code predictor).

I have a few questions:

What is the typical range of talker_loss and sub_talker_loss after the model converges well?
At what point would you consider the loss "too high" (i.e., something is wrong with training)?
Any reference numbers or training curves from your experiments would be very helpful. Thanks!

**Comment by OhMySimo** [comment]
I fine tuned it a month ago and I don't remember precisely but loss was somewhere around 6 when the model converged. At first steps the loss was like 14.

Two advices:
Don't use multiple speakers datasets, especially with mixed male and female speakers, and don't over-train (over training causes catastrophic forgetting and metaspeaker robotic voices)

Personally I got better results with Dia 1.6B for my language and datasets (Italian multispeaker) but also Qwen 3 did quite a good job with single speaker fine tuning.

Officially, fine tuned Qwen3 tts models are compatible only with "generate custom voice", but with a little edit of config.json I was able to use the fine tuned model as if it was the base one (in my case I needed to use generate_voice_clone).

Note that if you don't edit the training sft12hz script you'll also have to restore some keys from the original base model at the end of the fine tuning process to use generate_voice_clone etc...(I wrote a python script that copies missing keys from the original model.safetensors to the fine tuned one)

Also note that it's just a workaround and an architectural modification would be the best choice (personally I tried but didn't succeed)
